docs/source/custom_task.rst

```diff
@@ -1,37 +1,139 @@
-############
-Write a Task
-############
+Tutorial: Creating a Custom Task
+================================
 
-You can create your own Flash task to if you want to solve any other deep learning problem.
+In this tutorial we will go over the process of creating a custom task,
+along with a custom data module.
 
-See this example for defining a linear classifier task:
-
-.. testcode::
+.. code:: ipython3
 
+    import flash
+    
     import torch
-    import torch.nn.functional as F
-    from pytorch_lightning.metrics import Accuracy
-    from flash.core.classification import ClassificationTask
-
-    class LinearClassifier(ClassificationTask):
-        def __init__(
-            self,
-            num_inputs: int,
-            num_classes: int,
-            loss_fn=F.cross_entropy,
-            optimizer=torch.optim.SGD,
-            metrics=[Accuracy()],
-            learning_rate=1e-3,
-        ):
-            super().__init__(model=None,
+    from torch.utils.data import TensorDataset, DataLoader
+    from torch import nn
+    from sklearn import datasets
+    from sklearn.model_selection import train_test_split
+
+The Task
+--------
+
+Here we create a basic linear regression task by subclassing
+``flash.Task``. For the majority of tasks, you will likely only need to
+override the ``__init__`` and ``__forward__`` methods of task.
+
+.. code:: ipython3
+
+    class LinearRegression(flash.Task):
+        def __init__(self, num_inputs, learning_rate=0.001, metrics=None):
+            # what kind of model do we want?
+            model = nn.Linear(num_inputs, 1)
+    
+            # what loss function do we want?
+            loss_fn = torch.nn.functional.mse_loss
+            
+            # what optimizer to do we want?
+            optimizer = torch.optim.SGD
+            
+            super().__init__(
+                model=model,
                 loss_fn=loss_fn,
                 optimizer=optimizer,
                 metrics=metrics,
                 learning_rate=learning_rate,
             )
-            self.save_hyperparameters()
+            
+        def forward(self, x):
+            return self.model(x)
 
-            self.linear = torch.nn.Linear(num_inputs, num_classes)
+Where is the training step?
```

Comment:
Where **are** the training step?

Reply:
Where are the training/validation/test steps?